### PR TITLE
439 440 fabic policies l3 and physical domain

### LINKED ISSUE #439: Enhancement: New module for Physical Domains in Fabric Fabric Policies Template
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Create new module for Physical Domains in Tenant Fabric Policies Template ( created via ndo_template module )
part of #392 

* xxxx

### New or Affected Module(s):

* mso_XXXXX

### MSO version and MSO Platform

* ND 3.1.1 / NDO 4.3.1

### APIC version and APIC Platform for Site Level Resources

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Collection versions

* cisco.mso x.x.x

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #439 #440

### CODE REVIEW

**Review by sajagana — DISMISSED** [review]
LGTM!

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: plugins/modules/ndo_l3_domain.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    l3_domain: ansible_test_l3_domain
+    state: query
+  register: query_one
+
+- name: Query all l3 domains in the template
+  cisco.mso.ndo_l3_domain:
```
[the remark this answers is not in the document]
Comment (akinross) [reply plugins/modules/ndo_l3_domain.py]: changed

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by anvitha-jain — APPROVED** [review]

File: plugins/modules/ndo_vlan_pool.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    type: str
+  vlan_ranges:
+    description:
+    - A list of vlan ranges attached to the VLAN Pool.
+    - The list of configured vlan ranges must contain at least one entry.
+    - When the list of vlan ranges is null the update will not change existing entry configuration.
+    type: list
+    elements: dict
+    suboptions:
+      from_vlan:
```
Comment (samiib) [inline plugins/modules/ndo_vlan_pool.py:52]: Looks like the same change from here. https://github.com/CiscoDevNet/ansible-mso/pull/471#discussion_r1661719198

**Review by samiib — CHANGES_REQUESTED** [review]
I had a comment from the previous PR that hasn't merged yet. Not sure if you want to fix it here or in the original PR. 😵

**Review by samiib — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by samiib — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**